modules/multiplayer/editor/editor_network_profiler.cpp

```diff
@@ -72,43 +88,132 @@ void EditorNetworkProfiler::_update_frame() {
+				node_data[id] = NodeInfo(id);
+			}
+		}
+
+		TreeItem *node = replication_display->create_item(root);
+
+		const NodeInfo &root_info = node_data[E.value.root_node];
+		const NodeInfo &sync_info = node_data[E.value.synchronizer];
+		const NodeInfo &cfg_info = node_data[E.value.config];
+
+		node->set_text(0, root_info.path.get_file());
+		node->set_icon(0, has_theme_icon(root_info.type, SNAME("EditorIcons")) ? get_theme_icon(root_info.type, SNAME("EditorIcons")) : node_icon);
+		node->set_tooltip_text(0, root_info.path);
+
+		node->set_text(1, sync_info.path.get_file());
+		node->set_icon(1, get_theme_icon("MultiplayerSynchronizer", SNAME("EditorIcons")));
+		node->set_tooltip_text(1, sync_info.path);
+
+		int cfg_idx = cfg_info.path.find("::");
+		if (cfg_info.path.begins_with("res://") && ResourceLoader::exists(cfg_info.path) && cfg_idx > 0) {
+			String res_idstr = cfg_info.path.substr(cfg_idx + 2).replace("SceneReplicationConfig_", "");
+			String scene_path = cfg_info.path.substr(0, cfg_idx);
+			node->set_text(2, vformat("%s (%s)", res_idstr, scene_path.get_file()));
+			node->add_button(2, get_theme_icon(SNAME("InstanceOptions"), SNAME("EditorIcons")));
+			node->set_tooltip_text(2, cfg_info.path);
+			node->set_metadata(2, scene_path);
+		} else {
+			node->set_text(2, cfg_info.path);
+			node->set_metadata(2, "");
+		}
+
+		node->set_text(3, vformat("%d - %d", E.value.incoming_syncs, E.value.outgoing_syncs));
+		node->set_text(4, vformat("%d - %d", E.value.incoming_size, E.value.outgoing_size));
+	}
+}
+
+Array EditorNetworkProfiler::pop_missing_node_data() {
+	Array out;
+	for (const ObjectID &id : missing_node_data) {
+		out.push_back(id);
+	}
+	missing_node_data.clear();
+	return out;
+}
+
+void EditorNetworkProfiler::add_node_data(const NodeInfo &p_info) {
+	ERR_FAIL_COND(!node_data.has(p_info.id));
+	node_data[p_info.id] = p_info;
+	dirty = true;
+}
+
 void EditorNetworkProfiler::_activate_pressed() {
 	if (activate->is_pressed()) {
+		refresh_timer->start();
 		activate->set_icon(get_theme_icon(SNAME("Stop"), SNAME("EditorIcons")));
 		activate->set_text(TTR("Stop"));
 	} else {
+		refresh_timer->stop();
 		activate->set_icon(get_theme_icon(SNAME("Play"), SNAME("EditorIcons")));
 		activate->set_text(TTR("Start"));
 	}
 	emit_signal(SNAME("enable_profiling"), activate->is_pressed());
 }
 
 void EditorNetworkProfiler::_clear_pressed() {
-	nodes_data.clear();
+	rpc_data.clear();
+	sync_data.clear();
+	node_data.clear();
+	missing_node_data.clear();
 	set_bandwidth(0, 0);
-	if (frame_delay->is_stopped()) {
-		frame_delay->set_wait_time(0.1);
-		frame_delay->start();
+	refresh_rpc_data();
+	refresh_replication_data();
+}
+
+void EditorNetworkProfiler::_replication_button_clicked(TreeItem *p_item, int p_column, int p_idx, MouseButton p_button) {
+	if (!p_item) {
+		return;
+	}
+	String meta = p_item->get_metadata(p_column);
+	if (meta.size() && ResourceLoader::exists(meta)) {
+		EditorNode::get_singleton()->open_request(meta);
```

Comment:
Not blocking for this PR of course, but this is a good candidate for splitting it outside of EditorNode if we can, so that we don't have to include `editor_node.h` here. We should aim to put such helpers in smaller headers that don't get rebuilt all the time, to minimize incremental build time.

Reply:
I took a shortcut there indeed, this could be handled like a GDScript/Extension plugin would, via the `EditorInterface`. I'll have a look at refactoring it, since it should be trivial.